pkg/liquidity_provider.go

```diff
@@ -84,6 +86,67 @@ type CredentialsUpdateRequest struct {
 	NewPassword string `json:"newPassword" validate:"required"`
 }
 
+type GetReportsPeginPegoutRequest struct {
+	StartDate string `json:"startDate" validate:"required,datetime=2006-01-02"`
+	EndDate   string `json:"endDate" validate:"required,datetime=2006-01-02"`
+}
+
+func (r *GetReportsPeginPegoutRequest) ValidateGetReportsPeginPegoutRequest() error {
+	if r.StartDate == "" {
+		return errors.New("startDate is required")
+	}
+	if r.EndDate == "" {
+		return errors.New("endDate is required")
+	}
+
+	const dateFormat = "2006-01-02"
+	startDate, err := time.Parse(dateFormat, r.StartDate)
+	if err != nil {
+		return errors.New("startDate must be in format YYYY-MM-DD")
+	}
+
+	endDate, err := time.Parse(dateFormat, r.EndDate)
+	if err != nil {
+		return errors.New("endDate must be in format YYYY-MM-DD")
+	}
+
+	if endDate.Before(startDate) || endDate.Equal(startDate) {
+		return errors.New("endDate must be after startDate")
+	}
+
+	return nil
+}
+
+func (r *GetReportsPeginPegoutRequest) GetTimestamps() (startTime, endTime time.Time, err error) {
+	startTime, err = time.Parse("2006-01-02", r.StartDate)
+	if err != nil {
+		return time.Time{}, time.Time{}, err
+	}
+
+	endTime, err = time.Parse("2006-01-02", r.EndDate)
+	if err != nil {
+		return time.Time{}, time.Time{}, err
+	}
+
+	startTime = time.Date(
```

Comment:
what's the purpose of this? if later in the app you need the UTC timestamp you have a UTC method in time.Time

Reply:
same comment for the line below